src/transforms/uglify.js

```diff
@@ -1,18 +1,37 @@
-const {AST_Node, minify} = require('uglify-js');
-const {toEstree} = require('babel-to-estree');
-const types = require('babel-types');
-const walk = require('babylon-walk');
+const {minify} = require('uglify-es');
+const generate = require('babel-generator').default;
 
 module.exports = async function(asset) {
   await asset.parseIfNeeded();
 
-  // Convert to UglifyJS AST
-  var ast = AST_Node.from_mozilla_ast(toEstree(asset.ast, asset.contents));
-  var result = minify(ast, {
-    toplevel: true
-  });
+  // Convert AST into JS
+  let code = asset.isAstDirty
+    ? generate(asset.ast).code
+    : asset.outputCode || asset.contents;
```

Comment:
maybe just call `asset.generate().js` here?

Reply:
I extracted this bit from generate because i saw TODO: Source maps and thought this would affect performance in the future. I'll change it back to asset.generate though :)